bayesian/utility/v2/sa_models.py

```diff
@@ -69,3 +75,210 @@ def check_input_data(cls, values):  # noqa: F841 - Ignore 'cls' not used check
+    outlier_packages: List[Dict[str, Any]] = None
+    conflict_packages: List['ConflictPackages'] = None
+    current_stack_license: Dict[str, Any] = None
+    unknown_licenses: 'UnknownLicenses' = None
+    distinct_licenses: Optional[List[str]] = None
+
+
+class UsedByItem(BaseModel):  # noqa: D101
+    name: Optional[str] = None
+    stars: Optional[str] = None
+
+
+class GitHubDetails(BaseModel):  # noqa: D101
+    watchers: Optional[str] = None
+    first_release_date: Optional[str] = None
+    total_releases: Optional[str] = None
+    issues: Optional[Dict[str, Any]] = None
+    pull_requests: Optional[Dict[str, Any]] = None
+    dependent_repos: Optional[str] = None
+    open_issues_count: Optional[str] = None
+    latest_release_duration: Optional[str] = None
+    forks_count: Optional[str] = None
+    contributors: Optional[str] = None
+    size: Optional[str] = None
+    stargazers_count: Optional[str] = None
+    used_by: Optional[List[UsedByItem]] = None
+    dependent_projects: Optional[str] = None
+
+
+class PackageDetails(Package):  # noqa: D101
+    latest_version: str
+    github: Optional['GitHubDetails'] = None
+    licenses: Optional[List[str]] = None
+    ecosystem: 'Ecosystem'
+    url: Optional[str] = None
+
+
+class PackageDetailsForRegisteredUser(PackageDetails):  # noqa: D101
+    public_vulnerabilities: Optional[List['PremiumVulnerabilityFields']] = Field(
+        None, description='Publicly known vulnerability details'
+    )
+    private_vulnerabilities: Optional[List['PremiumVulnerabilityFields']] = Field(
+        None,
+        description='Private vulnerability details, available only to registered\nusers\n',
+    )
+    recommended_version: Optional[str] = Field(
+        None,
+        description=('Recommended package version which includes '
+                     'fix for both public and private vulnerabilities.\n'),
+    )
+    vulnerable_dependencies: Optional[List['PackageDetailsForRegisteredUser']] = Field(
+        None, description='List of dependencies which are vulnerable.\n'
+    )
+
+
+class PackageDetailsForFreeTierUser(PackageDetails):  # noqa: D101
+    public_vulnerabilities: Optional[List['BasicVulnerabilityFields']] = Field(
+        None, description='Publicly known vulnerability details'
+    )
+    private_vulnerabilities: Optional[List['BasicVulnerabilityFields']] = Field(
+        None, description='Private vulnerability details with limited info'
+    )
+    recommended_version: Optional[str] = Field(
+        None,
+        description='Recommended package version which includes fix for public vulnerabilities.\n',
+    )
+    vulnerable_dependencies: Optional[List['PackageDetailsForFreeTierUser']] = Field(
+        None, description='List of dependencies which are vulnerable.\n'
+    )
+
+
+class StackAggregatorResult(BaseModel):  # noqa: D101
+    version: str
+    started_at: str
+    ended_at: str
+    recommendation: 'StackRecommendation'
```

Comment:
Why Recommender is merged with Aggregator? Though it would generate similar JSON response, schematically it is incorrect.

You may need to define [StackAnalysesResult]( https://github.com/fabric8-analytics/fabric8-analytics-server/blob/master/api_specs/v2/stack_analyses.yaml#L556), [StackAnalysesResultForRegisteredUser](https://github.com/fabric8-analytics/fabric8-analytics-server/blob/master/api_specs/v2/stack_analyses.yaml#L602) and [StackAnalysesResultForFreeTier](https://github.com/fabric8-analytics/fabric8-analytics-server/blob/master/api_specs/v2/stack_analyses.yaml#L618) to match with the spec.